src/ORM/Connect/Database.php

```diff
@@ -20,6 +20,13 @@
 abstract class Database
 {
 
+    const PARTIAL_QUERY = 'partial_query';
+    const FULL_QUERY = 'full_query';
+    /**
+     * @var array
+     */
+    protected static $whitelist_array = [];
```

Comment:
does it have to be protected? could we make it private and add a getter?

Reply:
If it's supposed to be config it will need to be private, otherwise the status quo is to use protected over private (until #8996 is resolved 😉)

Reply:
I believe it's not the case if we'd have a getter for that :)
Otherwise, we'll need to mark it as `@internal` at least.

Reply:
However, since we have a public setter, I suggest we add a public getter too :)

Reply:
sweet, sorted by using `@internal`

Reply:
nice one